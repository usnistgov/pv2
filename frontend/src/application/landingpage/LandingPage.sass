@use 'src/colors'


.start-button
  margin: 64px auto
  text-align: center

.lading-page-container
  display: flex
  flex-direction: column

.landing-page-title
  width: fit-content
  align-self: center
  margin: 64px 0
  display: flex
  flex-direction: column

  h1
    align-self: center
    font-size: 48px
    margin-bottom: 0

  h2
    align-self: end
    margin-right: 4rem
    font-style: italic
    font-size: 20pt
    font-weight: lighter

.landing-page-section
  padding: 64px 0

  display: flex
  flex-direction: row

  .landing-content
    flex: 1
    box-sizing: border-box
    display: flex
    flex-direction: column
    justify-content: flex-start
    align-items: flex-start
    font-size: 18px
    text-align: justify
    gap: 0.5em

    margin: auto 16px

    hr
      width: 100%
      border: 1px solid colors.$primary

  .landing-image
    align-self: center
    flex: 1
    box-sizing: border-box
    margin: auto 16px

    img
      border-radius: 8px
      width: 100%
      height: 90%
      object-fit: contain

@media screen and (max-width: colors.$breakpoint-tablet)
  .landing-page-section
    padding: 2em

    flex-direction: column

    text-align: center

    .landing-content
      margin: 0
      align-items: center
      padding-bottom: 2em

    .landing-image
      margin: 0
