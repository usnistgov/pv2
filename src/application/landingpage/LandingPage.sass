@use 'src/colors'

.start-button
  margin: 64px auto
  text-align: center

.landing-page-title
  margin: 64px 0

  text-align: center

  h1
    font-size: 48px

.landing-page-section
  padding: 64px 0

  display: flex
  flex-direction: row

  .landing-content
    flex: 1
    box-sizing: border-box
    display: flex
    flex-direction: column
    justify-content: flex-start
    align-items: flex-start
    font-size: 18px
    gap: 0.5em

    margin: auto 16px

    hr
      width: 100%
      border: 1px solid colors.$primary

  .landing-image
    align-self: center
    flex: 1
    box-sizing: border-box
    margin: auto 16px

    img
      border-radius: 8px
      width: 100%
      height: 90%
      object-fit: contain

@media screen and (max-width: colors.$breakpoint-tablet)
  .landing-page-section
    padding: 2em

    flex-direction: column

    text-align: center

    .landing-content
      margin: 0
      align-items: center
      padding-bottom: 2em

    .landing-image
      margin: 0
