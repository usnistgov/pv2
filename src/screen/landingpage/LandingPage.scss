@import '../../main.scss';

.landing-page-wrapper {
  padding: 4vw 12vw;

  display: flex;
  flex-direction: row;

  .landing-content {
    flex: 1;
    box-sizing: border-box;
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    align-items: flex-start;
    font-size: 18px;
    gap: 0.5em;

    margin: 0em 2em;

    h1 {
      font-size: 48px;
    }

    hr {
      width: 100%;
      border: 1px solid $primary;
    }
  }

  // todo: figure out a good way to handle images/sizing
  .landing-image {
    flex: 1;
    box-sizing: border-box;
    margin: 2em;

    img {
      width: 100%;
      height: 90%;
      object-fit: contain;
    }
  }
}

@media screen and (max-width: $breakpoint-tablet) {
  .landing-page-wrapper {
    padding: 2em;

    flex-direction: column;

    text-align: center;

    .landing-content {
      margin: 0em;
      align-items: center;
      padding-bottom: 2em;
    }

    .landing-image {
      margin: 0em;
    }
  }
}
