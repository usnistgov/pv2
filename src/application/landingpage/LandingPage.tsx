import React, {ReactElement, useContext, useEffect} from "react";

// Library imports
import {Link} from "react-router-dom";
import {Button} from "@material-ui/core";
import {Icon as MdiIcon} from "@mdi/react";
import {mdiArrowRight, mdiArrowUp} from "@mdi/js";
import {Store} from "../ApplicationStore";

// Stylesheets
import './LandingPage.sass';

/**
 * The main component for the landing page. Displays some information about the app and a photo.
 */
export default function LandingPage(): ReactElement {
    const store = useContext(Store);
    useEffect(() => store.reset());
    return (
        <>
            <div className={"landing-page-title"} id={"title"}>
                <h1>Present Value of PhotoVoltaics – [PV]<sup>2</sup></h1>
            </div>
            <div className={"start-button"}>
                <Button
                    component={Link}
                    to={"/application"}
                    endIcon={<MdiIcon path={mdiArrowRight} size={1}/>}
                    variant={"contained"}
                    color={"primary"}
                    size={"large"}>
                    Start
                </Button>
            </div>
            <div className="landing-page-section">
                <div className="landing-content">
                    <hr/>
                    <div>
                        [PV]<sup>2</sup> is a web application that allows homeowners to determine a complete cost of
                        ownership for residential rooftop solar photovoltaic (PV) systems including purchase and
                        operation
                        through the system’s service life. The goal of [PV]<sup>2</sup> is to assist homeowners in
                        making
                        economical decisions related to solar PV. The current version is focused on allowing a homeowner
                        to evaluate the cost-effectiveness of a specific solar PV system from a proposed contact from a
                        solar installer. The homeowner will need information from the solar installer contract and their
                        electricity bill.
                    </div>
                    <hr/>
                </div>
                <div className="landing-image">
                    <img src={"/images/PV2 Logo.png"} alt={"PV2 logo"}/>
                </div>
            </div>
            <div className="landing-page-section">
                <div className="landing-image">
                    <img src={"/images/E3 Logo.png"} alt={"E3 logo"}/>
                </div>
                <div className="landing-content">
                    <hr/>
                    <div>
                        [PV]<sup>2</sup> uses the Economic Evaluation Engine (E3) API developed under the&nbsp;
                        <a href={"https://www.nist.gov/programs-projects/metrics-and-tools-sustainable-buildings"}>
                            Metrics and Tools for Sustainable Buildings Project
                        </a>
                        &nbsp;in the&nbsp;
                        <a href={"https://www.nist.gov/programs-projects/net-zero-energy-high-performance-buildings-program"}>
                            Net Zero Energy High Performance Building Program
                        </a>
                        &nbsp;overseen by the&nbsp;
                        <a href={"https://www.nist.gov/el/applied-economics-office"}>
                            Applied Economics Office (AEO)
                        </a>
                        &nbsp;in the&nbsp;
                        <a href={"https://www.nist.gov/el"}>
                            Engineering Laboratory (EL)
                        </a>
                        &nbsp;at the&nbsp;
                        <a href={"https://www.nist.gov/"}>
                            National Institute of Standards and Technology (NIST)
                        </a>
                        . E3 provides ASTM standards-based economic evaluation across a range of analysis types,
                        including
                        life cycle cost analysis (LCCA) and benefit-cost analysis (BCA).
                    </div>
                    <hr/>
                </div>
            </div>
            <div className="landing-page-section">
                <div className="landing-content">
                    <hr/>
                    <div>
                        For additional information on either [PV]<sup>2</sup> or E3, please contact&nbsp;
                        <a href={"https://www.nist.gov/people/joshua-d-kneifel"}>
                            Joshua Kneifel
                        </a>
                        &nbsp;at&nbsp;
                        <a href={"mailto:Joshua.Kneifel@nist.gov"}>
                            Joshua.Kneifel@nist.gov
                        </a>
                        .
                    </div>
                    <hr/>
                </div>
                <div className="landing-image" style={{"height": "500px"}}>
                    <img src={"/images/example.png"} alt={"example result"}/>
                </div>
            </div>
            <div className={"start-button"}>
                <Button
                    onClick={() => window.scrollTo({top: 0, behavior: 'smooth'})}
                    endIcon={<MdiIcon path={mdiArrowUp} size={1}/>}
                    variant={"contained"}
                    size={"large"}>
                    Top
                </Button>
            </div>
        </>
    );
}
