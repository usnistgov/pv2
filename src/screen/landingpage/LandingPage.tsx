import React from "react";
import './LandingPage.scss'
import {useHistory} from "react-router-dom";
import Button from "../../components/Button/Button";
import LandingImage from '../../assets/images/landing-image.png';
import Header from "../../components/Header/Header";

const LandingPage = () => {
  const history = useHistory();

  return (
    <>
      <Header />
      <div className="landing-page-wrapper">
        <div className="landing-content"> 
          <h1>PV2</h1>
          <hr />
          <div>Present Value of PhotoVoltaics (PV2) is an app that allows homeowners to determine a complete cost of ownership for residential rooftop solar PV systems, helping homeowners make economically wise decisions.</div>
          <hr />
          <div>PV2 uses the E3 API developed at NIST to run an economic analysis of rooftop solar photovoltaics.</div>
          <hr/>
          <Button text={"Try it out!"} onClick={() => history.push("/application")}/>
        </div>
        <div className="landing-image">
          <img src={LandingImage} alt={"landing page"} />
        </div>
      </div>
    </>
  );
}

export default LandingPage;