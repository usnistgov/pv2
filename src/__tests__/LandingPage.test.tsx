import React from 'react'
import { render } from '@testing-library/react'
import LandingPage from '../screen/LandingPage/LandingPage'
import { shallow } from 'enzyme'
import '../setupTests'

describe("<LandingPage />", () => {
  test('landing page should have button', async () => {
    const wrapper = shallow(<LandingPage />)
    expect(wrapper.find("Button")).toHaveLength(1)
  })
  
  test('landing page should have image', async () => {
    const wrapper = shallow(<LandingPage />)
    expect(wrapper.find("img")).toHaveLength(1)
  })
})
