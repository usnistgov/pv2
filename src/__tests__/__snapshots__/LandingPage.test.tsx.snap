// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> button same as snapshot 1`] = `ShallowWrapper {}`;

exports[`<LandingPage /> landing page same as snapshot 1`] = `ShallowWrapper {}`;
